kratos/python_scripts/save_restart_process.py

```diff
@@ -30,48 +30,24 @@ def __init__(self, model, params):
         ## Overwrite the default settings with user-provided parameters
         params.ValidateAndAssignDefaults(default_settings)
         params.RemoveValue("help")
-        self.params = params
-        self.model = model
 
-        if self.params["model_part_name"].GetString() == "":
-            raise Exception('No "model_part_name" was specified!')
+        model_part = model[params["model_part_name"].GetString()]
 
-    def ExecuteInitialize(self):
-        model_part = self.model[self.params["model_part_name"].GetString()]
-
-        is_mpi_execution = (model_part.GetCommunicator().TotalProcesses() > 1)
-
-        if is_mpi_execution:
-            import KratosMultiphysics.TrilinosApplication
-            from trilinos_restart_utility import TrilinosRestartUtility as Restart
+        if model_part.GetCommunicator().TotalProcesses() > 1: # mpi-execution
+            from KratosMultiphysics.TrilinosApplication.trilinos_restart_utility import TrilinosRestartUtility as RestartUtility
         else:
-            from restart_utility import RestartUtility as Restart
-
-        self.params.AddValue("input_filename", self.params["model_part_name"])
-        self.params.RemoveValue("model_part_name")
-
-        self.restart_utility = Restart(model_part, self.params)
+            from KratosMultiphysics.restart_utility import RestartUtility
 
-    def ExecuteBeforeSolutionLoop(self):
-        pass
+        params.AddValue("input_filename", params["model_part_name"])
+        params.RemoveValue("model_part_name")
```

Comment:
Note that this is the kind of thing that makes it hard to run several analyses sequentially: the output parameters would no longer be valid input (they no longer define the model part name).

Reply:
also this should probably be discussed in person, the restart utility does not need this parameter and I don't like the idea of having parameters in the defaults that do nothing...

I think / hope this could be solved properly with #2241